src/app.ts

```diff
@@ -1,20 +1,32 @@
 import * as express from "express";
 import * as logger from "morgan";
 import * as path from "path";
+import * as cors from "cors";
 
 import { errorHandler, errorNotFoundHandler } from "./middlewares/errorHandler";
+import { attachCORSResponse } from './middlewares/proxyCORS';
 
 // Routes
 import { index } from "./routes/index";
 // Create Express server
 export const app = express();
 
+/**
+ * @TODO Add more details such such as whitelists, methods options, etc
+ */
+app.use(cors());
+
 // Express configuration
 app.set("port", process.env.PORT || 3030);
 
 app.use(logger("dev"));
 
-app.use(express.static(path.join(__dirname, "../public")));
+// No Template Engine is Defined
+// app.use(express.static(path.join(__dirname, "../public")));
+
+// Attach CORS response
+app.use(attachCORSResponse);
```

Comment:
Is this middleware something that the `cors` middleware doesn't support?

Reply:
Oh yes that is my bad. All that middleware is doing exactly what the app.use(cors()) middleware is doing, which is adding "Access Control allow origin" to the response stream. I had coded that in first before using the express cors module/package.   It is just a good way to see internally what was happening but yes that can be deleted and the express cors middleware can be expanded upon to white list certain domains and methods, etc.